### PR TITLE
Fix target instance selection for integration tests run from test explorer

### PR DESCRIPTION
**Pull request by sharwell**
<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes

(either VSO or GitHub links)

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs [hunk 1]
```diff
@@ -234,6 +234,15 @@ private static IEnumerable<ISetupInstance> EnumerateVisualStudioInstances()
         private static ISetupInstance LocateVisualStudioInstance(ImmutableHashSet<string> requiredPackageIds)
         {
             var vsInstallDir = Environment.GetEnvironmentVariable("VSInstallDir");
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs]: :memo: This is not set when run from Test Explorer. I saw a value `VSAPPIDDIR` set, but I wasn't sure what that was. The value used below seemed to be the obvious one to handle this case.

File: src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs [hunk 2]
```diff
@@ -234,6 +234,15 @@ private static IEnumerable<ISetupInstance> EnumerateVisualStudioInstances()
...
+            if (vsInstallDir is null)
+            {
+                vsInstallDir = Environment.GetEnvironmentVariable("__UNITTESTEXPLORER_VSINSTALLPATH__");
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs]: 😕 When I tried to port this to tunnelvisionlabs/MouseFastScroll, I found this environment variable was not set. @ManishJayaswal What's the difference between `__UNITTESTEXPLORER_VSINSTALLPATH__` and `VSAPPIDDIR`, and why should I pick one over the other?

Comment (ManishJayaswal) [reply src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs]: hmmm I don't know. @AbhitejJohn - do you know which one should be used?

Comment (jasonmalinowski) [inline src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs]: Why not prefer this first? If I launch VS from a developer command prompt (which I do regularly), VSInstallDir will be set and this codepath never considered.

File: src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs]: If that translates to "visual studio app ID directory", that might indeed be the right thing to use. Not sure who to confirm that with though.

**Comment by sharwell** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by sharwell** [comment]
retest windows_release_vs-integration_prtest please

**Comment by sharwell** [comment]
retest windows_debug_spanish_unit32_prtest please

**Comment by sharwell** [comment]
@jinujoseph for approval

**Comment by sharwell** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by sharwell** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jinujoseph** [comment]
Approved to merge for 15.8.Preview4

**Comment by sharwell** [comment]
@jinujoseph realized after asking that this was a test-only change, but I still want to see integration tests pass on CI before merge.

**Pull request merged**